### PR TITLE
[Code Health] refactor: random number generation

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

- Reorganize the random number generation code into its own package.
- Refactor `computeClaimHash()`  as `Claim#GetHash()`.

## Issue

Refactor random number generation to improve reusability downstream.

- #579
- #402

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: pkg/crypto/rand/float_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	"encoding/binary"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	poktrand "github.com/pokt-network/poktroll/pkg/crypto/rand"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+)
+
+func TestRandProbability(t *testing.T) {
```
Comment (bryanchriswhite) [inline pkg/crypto/rand/float_test.go:13]:
```suggestion
func TestSeededFloat32(t *testing.T) {
```

File: pkg/crypto/rand/float_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	var numTrueSamples, numFalseSamples int
+
+	for idx := int64(0); idx < sampleSize; idx++ {
+		idxBz := make([]byte, binary.MaxVarintLen64)
+		binary.PutVarint(idxBz, idx)
+		randFloat, err := poktrand.SeededFloat32(idxBz)
+		require.NoError(t, err)
+
+		if randFloat < 0 || randFloat > 1 {
+			t.Fatalf("secureRandFloat64() returned out of bounds value: %f", randFloat)
```
Comment (bryanchriswhite) [inline pkg/crypto/rand/float_test.go:29]:
```suggestion
			t.Fatalf("SeededFloat32() returned out of bounds value: %f", randFloat)
```

File: pkg/crypto/rand/samples.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package rand
+
+import "math"
+
+// requiredSampleSize calculates the number of samples needed to achieve a desired confidence level
```
Comment (bryanchriswhite) [inline pkg/crypto/rand/samples.go:5]:
```suggestion
// RequiredSampleSize calculates the number of samples needed to achieve a desired confidence level
```

**Review by Olshansk — APPROVED** [review]
👌

**Review by red-0ne — APPROVED** [review]
LGTM

**Comment by bryanchriswhite** [comment]
E2E tests passed locally but failed to run in CI across multiple commits:

image

**Pull request merged**